iceoryx2/src/port/notifier.rs

```diff
@@ -165,27 +167,37 @@ impl<Service: service::Service> Notifier<Service> {
         let listener_list = &service.state().dynamic_storage.get().event().listeners;
         let dynamic_storage = Rc::clone(&service.state().dynamic_storage);
 
-        let dynamic_notifier_handle = match dynamic_storage.get().event().add_notifier_id(port_id) {
-            Some(handle) => handle,
-            None => {
-                fail!(from origin, with NotifierCreateError::ExceedsMaxSupportedNotifiers,
-                            "{} since it would exceed the maximum supported amount of notifiers of {}.",
-                            msg, service.state().static_config.event().max_notifiers);
-            }
-        };
-
-        let new_self = Self {
+        let mut new_self = Self {
             listener_connections: ListenerConnections::new(listener_list.capacity()),
             default_event_id,
             listener_list_state: unsafe { UnsafeCell::new(listener_list.get_state()) },
             dynamic_storage,
-            dynamic_notifier_handle,
+            dynamic_notifier_handle: None,
         };
 
         if let Err(e) = new_self.populate_listener_channels() {
             warn!(from new_self, "The new Notifier port is unable to connect to every Listener port, caused by {:?}.", e);
         }
 
+        std::sync::atomic::compiler_fence(Ordering::SeqCst);
```

Comment:
Interesting. After reading the documentation, a few question marks remain. Especially since the example should be broken on a weakly ordered CPU